csharp/PhoneNumbers/TimezoneMapper.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PhoneNumbers
+{
+    public class TimezoneMapper
+    {
+        private readonly IDictionary<long, string[]> map;
+        private readonly IDictionary<string, List<string[]>> dotnetmap;
+        private readonly ConcurrentDictionary<string, TimeZoneInfo> tziCache;
+        private readonly PhoneNumberUtil phoneUtil;
+
+        internal TimezoneMapper(IDictionary<long, string[]> source, IDictionary<string, List<string[]>> dotnetSource, IList<TimeZoneInfo> initZones)
```

Comment:
This doesn't seem to match the Google Java API. I was assuming that this was going to be a port of https://github.com/google/libphonenumber/blob/master/java/geocoder/src/com/google/i18n/phonenumbers/PhoneNumberToTimeZonesMapper.java and would at least need to match the API spec.

Reply:
Hi. It's true that I did not implement this with adherence to the Google API.
I can change that if that is the goal of this port and what people want.

As an aside, I didn't implement it that way because returning a list with one element "Etc/Unknown" seems more burdensome on the callers than an empty collection.

If you want me to re-do this with strict conformance, I am willing to do that. Please let me know.
Thanks for your comments.